### ISSUE TITLE
Peer to peer networking?

### ISSUE THREAD

**Issue by rog77** [s-macke/jor1k#154]
Would it be possible to bypass the need for a constant connection to a private websockets server if one is trying to communicate between Jor1k clients, by implementing PeerJS?

**Comment by s-macke** [comment]
You can use peerJS to establish a connection between the different clients. However the connection is based on ethernet frames. 
So you have to implement some services at some point. E. g. a DHCP Server to provide an IP address. Otherwise you have to chose some sort of random IP for each client. Also you have to implement some sort of switch for the ethernet frames based on the MAC address. 
All this is possible, but not in a few lines of code.

**Comment by benjamincburns** [comment]
I agree with Sebastian, but with one small addition. It could be fairly easy if you only wish to network together two instances of jor1k. Simply swap ethernet frames between the two VM instances as if you'd connected the two with an ethernet crossover cable. This will only allow direct communication between the two machines, however. You'd likely need to configure them with static IPs as well.

**Comment by proxy-m** [comment]
Related to: 
#155 [Redirect web traffic on jor1k], 
[#198 [Use networking without the websocket proxy on v86]](https://github.com/copy/v86/issues/198), 
[#23 [Network support on v86]](https://github.com/copy/v86/issues/23),
[#569 [Networking between two or more instances on v86]](https://github.com/copy/v86/issues/569).

**Comment by hello-smile6** [comment]
> Related to: #155 [Redirect web traffic on jor1k], [#198 [Use networking without the websocket proxy on v86]](https://github.com/copy/v86/issues/198), [#23 [Network support on v86]](https://github.com/copy/v86/issues/23), [#569 [Networking between two or more instances on v86]](https://github.com/copy/v86/issues/569).

Ooh!
